shpc/main/registry/__init__.py

```diff
@@ -152,7 +157,6 @@ def _sync(
             add_new=add_new,
             local=local,
         )
-        remote.cleanup()
```

Comment:
Any reason that you removed this? If it's a temporary clone I think we'd want to cleanup.

Reply:
Ah I see you moved it into the sync function! I think my preference is to have it up here - the reason is because it allows for using the underlying function without doing the cleanup. The cleanup is on the wrapper that the client exposes, and shouldn't require a developer user to have it done every time.

Reply:
I moved the `cleanup()` to the same place as the `clone()`, so that precisely `sync_from_remote` cleans after itself, and the developer users don't have to remember that they need to call `cleanup()` themselves every time they call `sync_from_remote`. Especially as there is currently no need for the clone to persist beyond `sync_from_remote`.

Reply:
Adding a cleanup boolean doesn't fully solve this, as the functions would also have to store somewhere (or return) the local registries that need to be cleaned up.

Reply:
Can you update me about your thinking here?

Reply:
I'm of the same opinion as before - a user is likely to call "sync" and we always want to clean up the remote. A develoepr user that wants more control can use sync_from_remote, and they would then have instantiated their own local/remote and can do as they please. So I think I'd like to see:

 - sync: should call remote.cleanup()
 - sync_from_remote: should not do any cleanup (and this is what we use in testing, it's on us)

Let me know your thinking, and if it's changed from the above.